_extensions/bluesky-comments/bluesky-resolve.lua

```diff
@@ -0,0 +1,107 @@
+local BlueskyAPI = {}
+
+-- Base Bluesky API URL
+local BASE_API_URL = "https://api.bsky.app"
+local BASE_APP_URL = "https://bsky.app"
+
+
+-- Extract handle and post ID from a Bluesky URL
+-- Example: https://bsky.app/profile/handle.bsky.social/post/1234
+---@param url string A Bluesky post URL
+---@return string handle The extracted user's handle
+---@return string postId The extracted post ID
+local function extractPostInfo(url)
+  local handle, postId = url:match("bsky%.app/profile/([^/]+)/post/([^/]+)")
+  if not handle or not postId then
+    error("Invalid Bluesky URL format")
+  end
+  return handle, postId
+end
+
+-- Global cache for resolved handles
+local BSKY_RESOLVED_HANDLES = _G.BSKY_RESOLVED_HANDLES or {}
+_G.BSKY_RESOLVED_HANDLES = BSKY_RESOLVED_HANDLES
+
+-- Resolve a handle to a DID.
+-- See <https://docs.bsky.app/docs/advanced-guides/resolving-identities>.
+---@param handle string The user's handle to be resolved
+---@return string did The resolved DID for the user
+function BlueskyAPI.resolveHandle(handle)
+  if BSKY_RESOLVED_HANDLES[handle] ~= nil then
+    return BSKY_RESOLVED_HANDLES[handle]
+  end
+
+  local url = string.format("%s/xrpc/com.atproto.identity.resolveHandle?handle=%s", BASE_API_URL, handle)
+
+  quarto.log.info("[bluesky-comments] Request: " .. url)
+  local mt, contents = pandoc.mediabag.fetch(url)
+  quarto.log.info("[bluesky-comments] Response: ", contents)
+
+  if not contents then
+    error("Failed to resolve handle: " .. handle)
```

Comment:
I think this should be maybe `quarto.log.error()` and, then, trigger with an `assert()` being false to stop the render process and correctly propagate the error.

Reply:
My approach so far has been to emit error messages but not break rendering. That was, in part, by following the previous design where those things happened in the browser.

That said, looking at the errors we do emit, it does seem like it'd be more useful to error earlier.

I definitely don't mind erroring for the person who just added the shortcode and is having trouble getting it to work. I've been burned in the past with hugo sites where re-building my site would occasionally fail because someone deleted a tweet that I had embedded via shortcode. 

Those errors tend to show up for long-forgotten posts when doing a full re-render. Fixing the error could be as easy as updating a profile (which would avoid re-computation if the post is frozen) or it could require updating the post content (which would trigger a re-render).

Anyway, I'll try out error throwing in a new PR.